hive/src/main/java/com/esri/hadoop/hive/BinUtils.java

```diff
@@ -0,0 +1,76 @@
+package com.esri.hadoop.hive;
+
+import com.esri.core.geometry.Envelope;
+
+public class BinUtils {
+	final long numCols;
+	final double extentMin;
+	final double extentMax;
+	final double binSize;
+	
+	public BinUtils(double binSize) {
+		this.binSize = binSize;
+		
+		// absolute max number of rows/columns we can have
+		long maxBinsPerAxis = (long) Math.sqrt(Long.MAX_VALUE);

```

Comment:
If the Java language standard/definition guarantees 64-bit `long`, then something like
`1L<<31`